packages/commerce-sdk-react/src/auth/index.ts

```diff
@@ -406,10 +427,21 @@ class Auth {
                 }
             }
         }
-        return await this.queueRequest(
-            () => helpers.loginGuestUser(this.client, {redirectURI: this.redirectURI}),
-            true
-        )
+        this.logWarningOnClient(this.silenceWarnings, slasSecretWarningMsg)
+        return this.clientSecret
+            ? await this.queueRequest(
+                  () =>
+                      helpers.loginGuestUserPrivate(
+                          this.client,
+                          {},
+                          {clientSecret: this.clientSecret}
+                      ),
+                  true
+              )
+            : await this.queueRequest(
+                  () => helpers.loginGuestUser(this.client, {redirectURI: this.redirectURI}),
+                  true
+              )
```

Comment:
🚨Style Police🚨

Non blocking but..

It might look more slick write this code more like this like this, where we aren't returning a conditional. 

```
const loginMethod = isPrivate ? 'loginGuestUserPrivate' : 'loginGuest'
const restArgs = isPrivate ? [this.client, ...] : [...]

return helpers[loginMethod](this.client, ...restArgs) 
```